qemu/tests/virtio_console.py

```diff
@@ -578,12 +579,156 @@ def test_loopback():
             logging.error(msg)
             test.fail(msg)
 
+    def get_virtio_port_by_name(vm, name):
+        """
+        Get virtio port object by name in VM.
+        :param vm: vm object in this test
+        :param name: name of the port
+        """
+        for device in vm.devices:
+            if isinstance(device, qdevices.QDevice):
+                if device.get_param("name") == name:
+                    return device
+        return None
+
+    def get_device_by_id(vm, device_id):
+        """
+        Get device object by id in VM.
+        :param vm: vm object in this test
+        :param device_id: id of the device
+        """
+        for device in vm.devices:
+            if isinstance(device, qdevices.QDevice):
+                if device.get_param("id") == device_id:
+                    return device
+        return None
+
+    def get_bus_of_port(vm, port_device):
```

Comment:
Can we simply get it by `bus = port_device.get_params("bus")`, and then remove the address suffix like `bus = re.sub(r".\d$", "", bus)`